pkg/fleet-manager/application/rollout_helper.go

```diff
@@ -0,0 +1,444 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package application
+
+import (
+	"context"
+	"fmt"
+	"io/fs"
+	"time"
+
+	flaggerv1b1 "github.com/fluxcd/flagger/pkg/apis/flagger/v1beta1"
+	"github.com/pkg/errors"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	applicationapi "kurator.dev/kurator/pkg/apis/apps/v1alpha1"
+	fleetapi "kurator.dev/kurator/pkg/apis/fleet/v1alpha1"
+	fleetmanager "kurator.dev/kurator/pkg/fleet-manager"
+	"kurator.dev/kurator/pkg/fleet-manager/manifests"
+	plugin "kurator.dev/kurator/pkg/fleet-manager/plugin"
+)
+
+const (
+	// kurator rollout labels
+	RolloutLabel = "kurator.dev/rollout"
+
+	// StatusSyncInterval specifies the interval for requeueing when synchronizing status. It determines how frequently the status should be checked and updated.
+	StatusSyncInterval = 30 * time.Second
+)
+
+func (a *ApplicationManager) fetchRolloutClusters(ctx context.Context,
+	app *applicationapi.Application,
+	kubeClient client.Client,
+	fleet *fleetapi.Fleet,
+	syncPolicy *applicationapi.ApplicationSyncPolicy,
+) (map[fleetmanager.ClusterKey]*fleetmanager.FleetCluster, error) {
+	log := ctrl.LoggerFrom(ctx)
+	destination := getPolicyDestination(app, syncPolicy)
+	ClusterInterfaceList, result, err := a.fetchFleetClusterList(ctx, fleet, destination.ClusterSelector)
+	if err != nil || result.RequeueAfter > 0 {
+		return nil, err
+	}
+
+	fleetclusters := make(map[fleetmanager.ClusterKey]*fleetmanager.FleetCluster, len(ClusterInterfaceList))
+	for _, cluster := range ClusterInterfaceList {
+		kclient, err := fleetmanager.ClientForCluster(kubeClient, fleet.Namespace, cluster)
+		if err != nil {
+			return nil, err
+		}
+
+		kind := cluster.GetObject().GetObjectKind().GroupVersionKind().Kind
+		fleetclusters[fleetmanager.ClusterKey{Kind: kind, Name: cluster.GetObject().GetName()}] = &fleetmanager.FleetCluster{
+			Secret:    cluster.GetSecretName(),
+			SecretKey: cluster.GetSecretKey(),
+			Client:    kclient,
+		}
+	}
+	log.Info("Successful to fetch destination clusters for Rollout")
+	return fleetclusters, nil
+}
+
+func (a *ApplicationManager) syncRolloutPolicyForCluster(ctx context.Context,
+	rolloutPolicy *applicationapi.RolloutConfig,
+	destinationClusters map[fleetmanager.ClusterKey]*fleetmanager.FleetCluster,
+	policyName string,
+) (ctrl.Result, error) {
+	log := ctrl.LoggerFrom(ctx)
+
+	label := map[string]string{
+		RolloutLabel: policyName,
```

Comment:
Take care, policyName length can exceed 63

Reply:
I add labels to make sure it is create by  Kurator. But labels have a limit of 63. So I use annotations instead of labels.